core/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/CreateSnapshotRequest.java

```diff
@@ -402,7 +403,7 @@ public CreateSnapshotRequest source(Map<String, Object> source) {
                 includeGlobalState = nodeBooleanValue(entry.getValue(), "include_global_state");
             }
         }
-        indicesOptions(IndicesOptions.fromMap((Map<String, Object>) source, IndicesOptions.lenientExpandOpen()));
+        indicesOptions(IndicesOptions.fromMap(source, IndicesOptions.strictExpandOpen()));
```

Comment:
I think this should be fixed by falling back to `indicesOptions` instead (which is equal to `IndicesOptions.strictExpandOpen()` by default)